### PR TITLE
Add unit tests for CCS

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: Ben Ye <yb532204897@gmail.com>

### What type of PR is this? 
_(bug/feature/cleanup/docs/design/test/chore/refactor..)_

### What this PR does / why we need it:

This pr adds unit tests for the new CCS functionalities and refactors some old unit tests because they are duplicate and not really testing cases correctly.

I mainly add tests for project reference because I don't see project claim changes much.

### Which issue(s) this PR fixes(can be a reference to existing issue or jira/bz):

Fixes https://issues.redhat.com/browse/OSD-5042

### Pre-checks:
- [x] manually tested latest changes against crc/k8s
- [ ] run the `make coverage` command to generate new calculated coverage

### CODE REVIEW

File: pkg/controller/projectreference/projectreference_adapter_test.go [hunk 1]
```diff
@@ -2,26 +2,43 @@ package projectreference_test
...
+	testStructs "github.com/openshift/gcp-project-operator/pkg/util/mocks/structs"
+)
+
+var (
+	fakeError            = errors.New("fakeError")
+	stopProcessingResult = util.OperationResult{
+		RequeueDelay:   0,
+		RequeueRequest: false,
+		CancelRequest:  true,
+	}
```
Comment (NautiluX) [inline pkg/controller/projectreference/projectreference_adapter_test.go:35]: should't we be able to use `util.StopProcessing()` directly?

File: pkg/controller/projectreference/projectreference_adapter_test.go [hunk 2]
```diff
@@ -2,26 +2,43 @@ package projectreference_test
...
+
+	continueProcessingResult = util.OperationResult{
+		RequeueDelay:   0,
+		RequeueRequest: false,
+		CancelRequest:  false,
+	}
```
Comment (NautiluX) [inline pkg/controller/projectreference/projectreference_adapter_test.go:41]: Probably `util.ContinueProcessing()` ?

File: pkg/controller/projectreference/projectreference_controller_test.go [hunk 3]
```diff
@@ -100,7 +102,7 @@ var _ = Describe("ProjectReference controller reconcilation", func() {
 		})
 	})
 
-	Context("When you cannot get credenitals", func() {
+	Context("When you cannot get credentials", func() {
```
Comment (NautiluX) [inline pkg/controller/projectreference/projectreference_controller_test.go:105]: Good typo catch :+1:

File: pkg/controller/projectreference/projectreference_controller_test.go [hunk 4]
```diff
@@ -278,69 +280,123 @@ var _ = Describe("ProjectReference controller reconcilation", func() {
...
+					mockGCPClient.EXPECT().GetIamPolicy(gomock.Any()).Return(&cloudresourcemanager.Policy{}, nil)
+					mockGCPClient.EXPECT().SetIamPolicy(gomock.Any()).Return(&cloudresourcemanager.Policy{}, nil)
+					mockGCPClient.EXPECT().CreateServiceAccountKey(gomock.Any()).Return(&iam.ServiceAccountKey{PrivateKeyData: "dGVzdAo="}, nil)
+					mockKubeClient.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.Any()).Return(fakeError).Times(1)
+					mockKubeClient.EXPECT().Create(gomock.Any(), gomock.Any()).Return(nil)
+					mockKubeClient.EXPECT().Status().Return(mockUpdater)
+					mockUpdater.EXPECT().Update(gomock.Any(), gomock.Any()).Return(nil)
+					_, err := reconciler.Reconcile(reconcile.Request{NamespacedName: projectReferenceName})
+					Expect(err).ToNot(HaveOccurred())
+				})
```
Comment (NautiluX) [inline pkg/controller/projectreference/projectreference_controller_test.go:328]: This feels a bit like code duplication from below, maybe you can find a way without.
All those Its duplicate a lot of `EXPECT` calls, maybe we can (ccs as well as non-ccs) maybe they can be unified into `BeforeEach` blocks?

**Review by NautiluX — CHANGES_REQUESTED** [review]

File: pkg/controller/projectreference/projectreference_controller_test.go (hunk 4 above)
Comment (yeya24) [reply pkg/controller/projectreference/projectreference_controller_test.go:328]: Will try, thanks!

File: pkg/controller/projectreference/projectreference_adapter_test.go (hunk 2 above)
Comment (yeya24) [reply pkg/controller/projectreference/projectreference_adapter_test.go:41]: Done.

File: pkg/controller/projectreference/projectreference_adapter_test.go (hunk 1 above)
Comment (yeya24) [reply pkg/controller/projectreference/projectreference_adapter_test.go:35]: Done.

File: pkg/controller/projectreference/projectreference_controller_test.go (hunk 4 above)
Comment (yeya24) [reply pkg/controller/projectreference/projectreference_controller_test.go:328]: I see there is kind of duplicate, but I don't find a good way to do it. How to put CCS and non-CCS into one `BeforeEach` block? 

We use a reference to pass `projectReference` so I must have two separate blocks for CCS and non-CCS.
``` go
mockKubeClient.EXPECT().Get(gomock.Any(), projectReferenceName, gomock.Any()).SetArg(2, *projectReference)
```

**Review by NautiluX — APPROVED** [review]

**Review by rafael-azevedo — APPROVED** [review]

**Pull request merged**